spec/index.md

```diff
@@ -413,11 +418,11 @@ Particular concepts to note from the spec are:
 
 -   [CBOR Self-Describe](https://www.rfc-editor.org/rfc/rfc8949#self-describe)
 
-#### CDDL {#cddl}
+#### CDDL \{#cddl\}
 
 The [Concise Data Definition Language (CDDL)](https://datatracker.ietf.org/doc/html/rfc8610) is a data description language for CBOR. It is used at various points throughout this document to describe how certain data structures are encoded with CBOR.
 
-## The system state tree {#state-tree}
+## The system state tree \{#state-tree\}
```

Comment:
This results in the tags appearing in the (current) rendering:

![Screenshot from 2024-04-02 19-00-51](https://github.com/dfinity/interface-spec/assets/31483726/b7725204-ec69-466a-ab9f-98fc6946447e)

Reply:
And the links do not work.

Reply:
Maybe the docusaurus version must be updated in `/docusaurus` for the rendering to work?

Reply:
updated to use html tags instead, since this seems to be a part of [extended markdown syntax](https://www.markdownguide.org/extended-syntax/)

Reply:
Now we've lost parts of the table of contents:
now
![Screenshot from 2024-04-03 07-17-06](https://github.com/dfinity/interface-spec/assets/31483726/9bb6fea7-eacd-43b2-a2ae-eab66b82303c)
vs before
![Screenshot from 2024-04-03 07-17-31](https://github.com/dfinity/interface-spec/assets/31483726/202e3947-91f2-4bbf-b349-0ea134183715)

Reply:
E.g., line 130 of spec/index.md is still
```
### Principals {#principal}
```
so I wonder why you had to change some of the section headings, but not others.

Reply:
I think I just missed some. This is the regex I used to find and replace: `^###\s(.+)\s\{#(.+)\}`, `<h3 id="$2">$1</h3>`

Updated now. Hopefully I caught everything this time.

Reply:
Now the table of contents is even more incomplete. It seems that `<h3>` is not recognized properly as far as the table of contents is concerned. You can find a preview here: https://p3j7v-5yaaa-aaaak-qckka-cai.icp0.io/docs

Reply:
do you know how the toc is generated? is there an option to make it recognize html headers as toc items?

Reply:
It doesn't seem possible (see [here](That is confusing because it is not clear to the reader before this sentence that it is both an upper and lower bound.)). I'll ask in the next spec meeting if people are fine with dropping the table of contents.

Reply:
the one other thing I can think of, and hopefully its not too much of a hack, is to preserve the original headings without the heading ID and then add an invisible html heading directly below it that declares the id. e.g.

```markdown
### my heading
<h3 id="my-heading" style="display: none;" />

Foo bar baZ
```